src/modules/kpi-reporting/controller.js

```diff
@@ -0,0 +1,50 @@
+'use-strict';
+const helpers = require('@envage/water-abstraction-helpers');
+const dataService = require('./services/data-service');
+
+const Boom = require('@hapi/boom');
+const mappers = require('./lib/mappers');
+
+const isDataNull = (data, errorMessage) => {
+  return data === null ? errorMessage : '';
+};
+
+/**
+ * This method requests all the KPI data sets maps them and returns the data for the KPI UI
+ * @param {*} request
+ * @returns {Object} an object containing all the data sets required for the KPI UI
+ */
+const getKPIData = async (request) => {
```

Comment:
This function is making a lot of syncronous calls that could be async using Promise.all

e.g.

```
const [registration, delegatedAccess, etc] = await Promise.all([
  dataService.getIDMRegistrationsData(),
  dataService.getCRMDelegatedAccessData()
])

const errors = [
   getErrorMessage(registrations, 'registrations'),
   getErrorMessage(delegatedAccess, 'delegatedAccess'),
].filter(i => i)

if (errors.length) {
  return Boom.notFound(errors.join(', ')
}
```

Reply:
all done thanks